From 25a32c84a4fcd59bbd8c08f801907f62fcf124b8 Mon Sep 17 00:00:00 2001
From: Francesco Pham <francesco.pham1@h-partners.com>
Date: Mon, 3 Aug 2026 14:17:30 +0000
Subject: [PATCH] khdf: probe -mno-outline-atomics with cc-option

-mno-outline-atomics is an aarch64-only flag; passing it unconditionally
breaks x86_64 kernel builds (x86_general emulator) where clang raises
-Werror,-Woption-ignored. cc-option adds it only when the compiler
accepts it, leaving arm64 builds unchanged.

Change-Id: Icb07f9ab56bd7b3102f5989b87d74c49a57e0b54
Signed-off-by: Francesco Pham <francesco.pham1@h-partners.com>
---
 adapter/khdf/linux/manager/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/adapter/khdf/linux/manager/Makefile b/adapter/khdf/linux/manager/Makefile
index 7b09661e..b1ffb950 100644
--- a/adapter/khdf/linux/manager/Makefile
+++ b/adapter/khdf/linux/manager/Makefile
@@ -69,7 +69,7 @@ obj-y += \
         $(SEC_FUNCTION_OBJECTS)
 
 CFLAGS_REMOVE_$(HDF_FRAMEWORK_SRC_DIR)/framework/utils/src/hdf_sbuf.o += -mgeneral-regs-only
-KBUILD_CFLAGS += -mno-outline-atomics
+KBUILD_CFLAGS += $(call cc-option,-mno-outline-atomics)
 
 ccflags-y += \
        -I$(srctree)/drivers/hdf/framework/include \
-- 
2.50.1

